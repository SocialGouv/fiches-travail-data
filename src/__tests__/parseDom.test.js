import fs from "fs";
import { JSDOM } from "jsdom";
import path from "path";

const { parseDom } = require("../index");

const sample = fs.readFileSync(path.join(__dirname, "article.html")).toString();

test("should parse HTML section", () => {
  const dom = new JSDOM(sample);
  const parsed = parseDom(dom);
  expect(parsed).toMatchSnapshot();
});
