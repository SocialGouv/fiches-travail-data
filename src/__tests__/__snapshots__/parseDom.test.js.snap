// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse HTML section 1`] = `
Object {
  "date": "28/11/2018",
  "description": "Naissance, mariage, PACS, décès d'un proche… : lorsque ces événements surviennent, le salarié peut s'absenter pendant une durée minimale de 1 à 5 jours.",
  "intro": "<p>Naissance, mariage, conclusion d’un PACS, décès d’un proche…&nbsp;: lorsque ces événements surviennent, le salarié peut s’absenter pendant une durée minimale de 1 à 5 jours, selon les circonstances. A la demande d’absence adressée à l’employeur, doit être jointe une justification (certificat de naissance, de décès,…) de l’événement.<br class=\\"autobr\\">
Rémunérées, ces absences sont assimilées à du travail effectif pour la détermination des congés payés.</p>",
  "pubId": "article375531",
  "sections": Array [
    Object {
      "anchor": "",
      "description": "A SAVOIR 
Les dispositions présentées ici tiennent compte de la loi du 8 août 2016 citée en référence, en vigueur depuis le 10 août 2016. Selon cette loi, sont désormais distingués les domaines d’ordre public où le législateur fixe des règles auxquelles il n’est pas possible de déroger, les domaines qui sont les domaines où les règles pourront être fixées par accord collectif,  et enfin les règles fixées dites « supplétives » c’est à dire applicables à défaut d’accord collectif.",
      "html": "<div class=\\"texteencadre-spip spip\\"><strong>A SAVOIR</strong> <br class=\\"autobr\\">
Les dispositions présentées ici tiennent compte de la loi du 8 août 2016 citée en référence, en vigueur depuis le 10 août 2016. Selon cette loi, sont désormais distingués les domaines d’ordre public où le législateur fixe des règles auxquelles il n’est pas possible de déroger, les domaines qui sont les domaines où les règles pourront être fixées par accord collectif,  et enfin les règles fixées dites «&nbsp;supplétives&nbsp;» c’est à dire applicables à défaut d’accord collectif.</div>",
      "references": Object {},
      "text": " A SAVOIR 
Les dispositions présentées ici tiennent compte de la loi du 8 août 2016 citée en référence, en vigueur depuis le 10 août 2016. Selon cette loi, sont désormais distingués les domaines d’ordre public où le législateur fixe des règles auxquelles il n’est pas possible de déroger, les domaines qui sont les domaines où les règles pourront être fixées par accord collectif,  et enfin les règles fixées dites « supplétives » c’est à dire applicables à défaut d’accord collectif.",
      "title": "Les congés pour événements familiaux",
    },
    Object {
      "anchor": "Quels-sont-les-evenements-ouvrant-droit-a-conge",
      "description": "Le salarié a droit, sur justification, à un congé :",
      "html": "<p>Le salarié a droit, sur justification, à un congé&nbsp;:</p><ol class=\\"spip\\"><li> Pour son mariage ou pour la conclusion d’un pacte civil de solidarité (PACS)&nbsp;;</li><li> Pour le mariage d’un enfant&nbsp;;</li><li> Pour chaque naissance survenue à son foyer ou pour l’arrivée d’un enfant placé en vue de son adoption. Ces jours d’absence ne se cumulent pas avec les congés accordés pour ce même enfant dans le cadre du congé de maternité&nbsp;;</li><li> Pour le décès d’un enfant,</li><li> Pour le décès  du conjoint, du concubin ou du partenaire lié par un PACS, du père, de la mère, du beau-père, de la belle-mère, d’un frère ou d’une sœur&nbsp;;</li><li> Pour l’annonce de la survenue d’un handicap chez un enfant.</li></ol><blockquote class=\\"spip\\">
<p>Les dispositions mentionnées ci-dessus sont d’ordre public, de sorte qu’aucun accord collectif ou convention collective ne peut avoir pour effet, ni de priver le salarié du droit à s’absenter lorsque survient l’un de ces événements, ni de restreindre l’exercice de ce droit. Ainsi, par exemple, la loi ne prévoyant aucune condition d’ancienneté pour bénéficier de ces congés, aucun accord collectif ou aucune convention collective ne peut valablement imposer une telle condition.</p>
</blockquote>",
      "references": Object {},
      "text": "Le salarié a droit, sur justification, à un congé :Pour son mariage ou pour la conclusion d’un pacte civil de solidarité (PACS) ; Pour le mariage d’un enfant ; Pour chaque naissance survenue à son foyer ou pour l’arrivée d’un enfant placé en vue de son adoption. Ces jours d’absence ne se cumulent pas avec les congés accordés pour ce même enfant dans le cadre du congé de maternité ; Pour le décès d’un enfant, Pour le décès  du conjoint, du concubin ou du partenaire lié par un PACS, du père, de la mère, du beau-père, de la belle-mère, d’un frère ou d’une sœur ; Pour l’annonce de la survenue d’un handicap chez un enfant.Les dispositions mentionnées ci-dessus sont d’ordre public, de sorte qu’aucun accord collectif ou convention collective ne peut avoir pour effet, ni de priver le salarié du droit à s’absenter lorsque survient l’un de ces événements, ni de restreindre l’exercice de ce droit. Ainsi, par exemple, la loi ne prévoyant aucune condition d’ancienneté pour bénéficier de ces congés, aucun accord collectif ou aucune convention collective ne peut valablement imposer une telle condition.",
      "title": "Quels sont les événements ouvrant droit à congé ?",
    },
    Object {
      "anchor": "Quelle-est-la-duree-du-conge",
      "description": "La loi fixe une durée minimale (cf. ci-dessous). Une durée plus élevée peut être fixée pour chacun des congés légaux pour événements familiaux par la convention ou l’accord collectif d’entreprise ou, ",
      "html": "<p>La loi fixe une durée minimale (cf. ci-dessous). Une durée plus élevée peut être fixée pour chacun des congés légaux pour événements familiaux par la convention ou l’accord collectif d’entreprise ou, à défaut, la convention ou l’accord de branche. <br class=\\"autobr\\">
La durée minimale fixée par la loi est la suivante Cet accord ou cette convention ne peut toutefois fixer une durée inférieure à&nbsp;:</p><ol class=\\"spip\\"><li> Quatre jours pour le mariage du salarié ou pour la conclusion d’un PACS&nbsp;;</li><li> Un jour pour le mariage d’un enfant&nbsp;;</li><li> Trois jours pour chaque naissance survenue au foyer du salarié ou pour l’arrivée d’un enfant placé en vue de son adoption&nbsp;;</li><li> Cinq jours pour le décès d’un enfant&nbsp;;</li><li> Trois jours pour le décès du conjoint, du concubin ou du partenaire lié par un PACS, du père, de la mère, du beau-père, de la belle-mère, d’un frère ou d’une sœur&nbsp;;</li><li> Deux jours pour l’annonce de la survenue d’un handicap chez un enfant.</li></ol><blockquote class=\\"spip\\">
<p>A défaut de convention ou d’accord applicable dans son entreprise, le salarié a droit aux congés pour les événements familiaux et pour les durées mentionnés ci-dessus.</p>
</blockquote><p>Ces journées d’absence sont comptées en jours «&nbsp;ouvrables&nbsp;» (tous les jours de la semaine, sauf le jour de repos hebdomadaire - en principe, le dimanche - et les jours fériés habituellement non travaillés dans l’entreprise).</p>",
      "references": Object {},
      "text": "La loi fixe une durée minimale (cf. ci-dessous). Une durée plus élevée peut être fixée pour chacun des congés légaux pour événements familiaux par la convention ou l’accord collectif d’entreprise ou, à défaut, la convention ou l’accord de branche. 
La durée minimale fixée par la loi est la suivante Cet accord ou cette convention ne peut toutefois fixer une durée inférieure à :Quatre jours pour le mariage du salarié ou pour la conclusion d’un PACS ; Un jour pour le mariage d’un enfant ; Trois jours pour chaque naissance survenue au foyer du salarié ou pour l’arrivée d’un enfant placé en vue de son adoption ; Cinq jours pour le décès d’un enfant ; Trois jours pour le décès du conjoint, du concubin ou du partenaire lié par un PACS, du père, de la mère, du beau-père, de la belle-mère, d’un frère ou d’une sœur ; Deux jours pour l’annonce de la survenue d’un handicap chez un enfant.A défaut de convention ou d’accord applicable dans son entreprise, le salarié a droit aux congés pour les événements familiaux et pour les durées mentionnés ci-dessus.Ces journées d’absence sont comptées en jours « ouvrables » (tous les jours de la semaine, sauf le jour de repos hebdomadaire - en principe, le dimanche - et les jours fériés habituellement non travaillés dans l’entreprise).",
      "title": "Quelle est la durée du congé ?",
    },
    Object {
      "anchor": "Quelle-est-la-procedure",
      "description": "Pour bénéficier de l’autorisation d’absence au titre d’un des événements familiaux mentionnés ci-dessus (mariage, naissance, …), le salarié doit justifier de la survenance de l’événement en cause : le",
      "html": "<p>Pour bénéficier de l’autorisation d’absence au titre d’un des événements familiaux mentionnés ci-dessus (mariage, naissance, …), le salarié doit justifier de la survenance de l’événement en cause&nbsp;: le code du travail n’imposant aucun formalisme particulier, le salarié peut donc apporter cette justification par tous moyens.</p><p>Les congés doivent être pris au moment des événements en cause, le(s) jour(s) d’autorisation d’absence n’ayant pas à être nécessairement pris le jour de l’événement le justifiant, mais dans la période entourant cet événement. Ainsi, par exemple, en cas de mariage d’un enfant, le jour de congé peut être posé pour la date du mariage, mais aussi la veille de cette date ou le lendemain.</p><blockquote class=\\"spip\\">
<p>Les congés légaux pour événements familiaux n’entraînent pas de réduction de la rémunération et sont assimilés à du temps de travail effectif pour la détermination de la durée du congé payé annuel. La durée de ces congés ne peut être imputée sur celle du congé payé annuel.<br class=\\"autobr\\">
En cas de différend, le refus de l’employeur peut être directement contesté par le salarié devant le conseil de prud’hommes, statuant en la forme des référés. Le conseil de prud’hommes statue alors en dernier ressort&nbsp;: seul un pourvoi en cassation est donc possible (cette disposition est issue du décret du 18 novembre 2016 cité en référence, dont la date d’entrée en vigueur a été fixée au 1er janvier 2017).</p>
</blockquote>",
      "references": Object {},
      "text": "Pour bénéficier de l’autorisation d’absence au titre d’un des événements familiaux mentionnés ci-dessus (mariage, naissance, …), le salarié doit justifier de la survenance de l’événement en cause : le code du travail n’imposant aucun formalisme particulier, le salarié peut donc apporter cette justification par tous moyens.Les congés doivent être pris au moment des événements en cause, le(s) jour(s) d’autorisation d’absence n’ayant pas à être nécessairement pris le jour de l’événement le justifiant, mais dans la période entourant cet événement. Ainsi, par exemple, en cas de mariage d’un enfant, le jour de congé peut être posé pour la date du mariage, mais aussi la veille de cette date ou le lendemain.Les congés légaux pour événements familiaux n’entraînent pas de réduction de la rémunération et sont assimilés à du temps de travail effectif pour la détermination de la durée du congé payé annuel. La durée de ces congés ne peut être imputée sur celle du congé payé annuel.
En cas de différend, le refus de l’employeur peut être directement contesté par le salarié devant le conseil de prud’hommes, statuant en la forme des référés. Le conseil de prud’hommes statue alors en dernier ressort : seul un pourvoi en cassation est donc possible (cette disposition est issue du décret du 18 novembre 2016 cité en référence, dont la date d’entrée en vigueur a été fixée au 1er janvier 2017).",
      "title": "Quelle est la procédure ?",
    },
  ],
  "title": "Les congés pour événements familiaux",
}
`;
